jacquard/experiments/constraints.py

```diff
@@ -216,3 +216,23 @@ def matches_user(self, user, context=None):
             return False
 
         return True
+
+    def is_disjoint_from_constraints(self, other_constraints):
+        """
+        Test whether the set of users matching ``self``, and the set matching
+        ``constraints`` are disjoint.
+
+        Currently only constraints on anonymity and tags are supported, era,
+        and ``joined_before``/``joined_after`` constraints are not supported.
+        """
+        if self.include_anonymous and other_constraints.include_anonymous:
+            return False
+
+        if (
+            set(self.required_tags) & set(other_constraints.excluded_tags)
```

Comment:
Mild preference for `set.isdisjoint(self.required_tags, other_constraints.excluded_tags)` here.

Reply:
7ea350c